src/libAtomVM/bif.c

```diff
@@ -494,6 +494,17 @@ term bif_erlang_add_2(Context *ctx, uint32_t fail_label, int live, term arg1, te
     }
 }
 
+term bif_erlang_add_1(Context *ctx, uint32_t fail_label, int live, term arg1)
+{
+    UNUSED(live);
+
+    if (LIKELY(term_is_integer(arg1))) {
+        return arg1;
+    } else {
+        RAISE_ERROR_BIF(fail_label, BADARITH_ATOM);
```

Comment:
Could we have a test for the barith case?
What is the expected behavior for floats? (a test would help here as well to compare with BEAM).